bugfixtemplates/synchronized-object-reassignment/SynchronizedObjectReassignment.java

```diff
@@ -0,0 +1,48 @@
+package bugfixtemplates;
+
+class SynchronizedObjectReassignment {
+
+    Object mutex, mutex1;
+
+    void m0() {
+        mutex = new Object();
+        foo(mutex);
+    }
+
+    void m1() {
+        synchronized (mutex) {
+            mutex = new Object();
+            foo(mutex);
+        }
+    }
+
+    void m2() {
+        synchronized (mutex) {
+            mutex1 = new Object();
+            foo(mutex);
+        }
+    }
+
+    void m3() {
+        synchronized (mutex) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

Object mutex synchronized on line 27 should not be re-assigned in the same synchronized block on line 28. After this re-assignment, you lose the reference to the synchronized object.